core/primitives/src/account.rs

```diff
@@ -24,6 +28,45 @@ impl Account {
     }
 }
 
+/// Calculates the storage and the name rent for the given account
+/// for period of account.storage_paid_at to current block_height
+pub fn calculate_rent(
+    account_id: &AccountId,
+    account: &Account,
+    // TODO #1903 block_height: BlockHeight,
+    block_index: BlockHeight,
+    account_length_baseline_cost_per_block: Balance,
+    storage_cost_byte_per_block: Balance,
+) -> Balance {
+    let charge = u128::from(block_index - account.storage_paid_at)
+        * rent_per_block(
+            account_id,
+            account,
+            account_length_baseline_cost_per_block,
+            storage_cost_byte_per_block,
+        );
+    let actual_charge = std::cmp::min(account.amount, charge);
```

Comment:
We should probably expose the full unpaid rent here and handle overflow in other places.